packages/neuron-ui/src/components/SpecialAssetList/hooks.ts

```diff
@@ -91,3 +88,99 @@ export const useGetAssetAccounts = (walletID: string) => {
       .catch((err: Error) => console.error(err))
   }, [walletID, dispatch])
 }
+
+interface SpecialAssetColumnInfoProps {
+  cell: CKBComponents.CellOutput & {
+    data: string
+  }
+  datetime: number
+  epoch: string
+  assetInfo: AssetInfo
+  bestKnownBlockTimestamp: number
+  tokenInfoList: Array<Controller.GetTokenInfoList.TokenInfo>
+}
+
+export const useGetSpecialAssetColumnInfo = ({
+  cell: { capacity, lock, type, data },
+  datetime,
+  epoch,
+  assetInfo,
+  bestKnownBlockTimestamp,
+  tokenInfoList,
+}: SpecialAssetColumnInfoProps) => {
+  const [t] = useTranslation()
+
+  let targetTime: undefined | number
+  let status:
+    | 'user-defined-asset'
+    | 'locked-asset'
+    | 'claim-asset'
+    | 'withdraw-asset'
+    | 'transfer-nft'
+    | 'user-defined-token' = 'user-defined-asset'
+  let epochsInfo: Record<'target' | 'current', number> | undefined
+  let amount: string = `${shannonToCKBFormatter(capacity)} CKB`
+
+  switch (assetInfo.lock) {
+    case PresetScript.Locktime: {
+      const targetEpochInfo = epochParser(ckbCore.utils.toUint64Le(`0x${lock.args.slice(-16)}`))
+      const currentEpochInfo = epochParser(epoch)
+      const targetEpochFraction =
+        Number(targetEpochInfo.length) > 0 ? Number(targetEpochInfo.index) / Number(targetEpochInfo.length) : 1
+      epochsInfo = {
+        target: Number(targetEpochInfo.number) + Math.min(targetEpochFraction, 1),
+        current: Number(currentEpochInfo.number) + Number(currentEpochInfo.index) / Number(currentEpochInfo.length),
+      }
+      targetTime = bestKnownBlockTimestamp + (epochsInfo.target - epochsInfo.current) * MS_PER_EPOCHS
+      if (epochsInfo.target - epochsInfo.current > 0) {
+        status = 'locked-asset'
+      } else {
+        status = 'claim-asset'
+      }
+      break
+    }
+    case PresetScript.Cheque: {
+      status = (assetInfo as ChequeAssetInfo).data === 'claimable' ? 'claim-asset' : 'withdraw-asset'
+
+      if (status === 'withdraw-asset') {
+        const DAY = 86_400_000
```

Comment:
The constant name is a bit confusing, could be `MILLISECONDS_PER_DAY`, which is consistent with `MS_PER_EPOCHS` above

Reply:
It can be defined next to `MILLISECONDS_IN_YEAR` https://github.com/nervosnetwork/neuron/blob/develop/packages/neuron-ui/src/utils/const.ts#L25 so `MILLISECONDS_IN_YEAR` can be calculated based on `MILLISECONDS_PER_DAY`